packages/vincent-tools/vincent-tool-sdk/src/lib/get-onchain-policy-params.ts

```diff
@@ -0,0 +1,97 @@
+import { z } from "zod";
+import { ethers } from "ethers";
+
+import type { EthersAbiDecodedValue, Policy, VincentPolicyDef } from "./types";
+import { abiDecodePolicyParameters } from "./abi-decode-policy-params";
+import { formatZodError } from "./format-zod-error";
+
+interface AllOnChainPolicyParams {
+    isPermitted: boolean;
+    appId: ethers.BigNumber;
+    appVersion: ethers.BigNumber;
+    policies: Policy[];
+}
+
+export const getOnChainPolicyParams = async (args: {
+    yellowstoneRpcUrl: string,
+    vincentContractAddress: string,
+    appDelegateeAddress: string,
+    agentWalletPkpTokenId: string,
+    toolIpfsCid: string,
+    policyIpfsCid: string,
+    policyUserParamsSchema?: z.infer<VincentPolicyDef['userParamsSchema']>
+}): Promise<z.infer<VincentPolicyDef['userParamsSchema'] | undefined>> => {
+    try {
+        const allOnChainPolicyParams = await _getAllOnChainPolicyParams(args);
```

Comment:
This method throws its own very descriptive error and we don't need to wrap it -- move outside of `try{}` IMO